backend/src/apiserver/storage/object_store.go

```diff
@@ -151,6 +152,27 @@ func (m *MinioObjectStore) GetSignedUrl(bucketConfig *objectstore.Config, secret
 	return signedUrl.String(), nil
 }
 
+func (m *MinioObjectStore) GetSignedUrlWithQueryParams(bucketConfig *objectstore.Config, secret *v1.Secret, expirySeconds time.Duration, artifactURI string, queryParams url.Values) (string, error) {
```

Comment:
Could you make `GetSignedURL` call this method with empty queryParams so we don't duplicate code?